cmake/config/template-arguments.in

```diff
@@ -208,17 +208,24 @@ GENERAL_CONTAINER_TYPES := { std::vector;
                              ArrayView
                            }
 
+// DoFHandler (for iterators!)
+DOFHANDLER_TEMPLATE := { DoFHandler }
+
 // DoFHandler and hp::DoFHandler templates
-DOFHANDLER_TEMPLATES := { DoFHandler;
-                          hp::DoFHandler }
+DOFHANDLER_TEMPLATES := { DoFHandler; hp::DoFHandler }
```

Comment:
There are multiple functions templated with `DoFHandlerType`. These require for the time being that they are instantiated for `DoFHandler` and for the dummy `DoFHandler`. I have decided against eliminating all `DoFHandlerType`, since then the number of changed lines would be even larger. (At the beginning, I did it, but then changed the strategy). I'll open in a sec an issue to collect follow-up tasks.